### PR TITLE
Submission service

### PR DESCRIPTION
**Pull request by westonkd**
_Closes PLAT-2545_
Test Plan:
Verify you can still fetch submission details from Canvas after receiving a webhook.

### CODE REVIEW

File: app/controllers/event_controller.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 class EventController < ApplicationController
   skip_before_action :verify_authenticity_token
 
+  # submission
+  #
+  # Creates a submission for the specified assignment when a
+  # webhook is reived.
```
Comment (abutterf) [inline app/controllers/event_controller.rb]: s/reived/received

File: app/controllers/submissions_controller.rb [hunk 2]
```diff
@@ -6,6 +6,19 @@ class SubmissionsController < ApplicationController
...
+  #
+  # retrieves the following submission data from Canvas
+  # after a submission is received via webhook:
+  #
+  # assignment_id - before this point the actual ID of the
+  # assignment is not known. This data is contained in the
+  # payload of the Canvas submission endpoint which is used
+  # in this method.
+  #
+  # attachments - data regaurding the attachments is needed to
```
Comment (abutterf) [inline app/controllers/submissions_controller.rb]: s/regaurding/reguarding

File: app/helpers/registration_helper.rb [hunk 3]
```diff
@@ -24,12 +35,25 @@ def registration_services
...
   def tool_proxy_registration_service
     @_tool_proxy_registration_service ||= begin
       registration_services.registration_service
     end
   end
 
+  # create_tool_proxy
+  #
+  # Attempts to create the tool proxy in Canvas.
+  # If succesful the tool consumer's half the shared
```
Comment (abutterf) [inline app/helpers/registration_helper.rb]: s/succesful/successful
the tool consumer's half of the shared

File: app/helpers/registration_helper.rb [hunk 4]
```diff
@@ -39,22 +63,51 @@ def create_tool_proxy(tool_proxy)
     false
   end
 
+  # authorization_service
+  #
+  # Returns the service used to retrive access
```
Comment (abutterf) [inline app/helpers/registration_helper.rb]: s/retrive/retrieve

File: app/helpers/submissions_helper.rb [hunk 5]
```diff
@@ -1,9 +1,15 @@
 module SubmissionsHelper
   include LtiAuthorizationHelper
 
+  # retrieve_submission
+  #
+  # Retrive a submission and its details via the submission
```
Comment (abutterf) [inline app/helpers/submissions_helper.rb]: s/Retrive/Retrieve

File: db/migrate/20170424144902_add_submission_service_url_to_tool_proxy.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class AddSubmissionServiceUrlToToolProxy < ActiveRecord::Migration[5.0]
+  def change
+    add_column :tool_proxies, :submission_service_url, :string
+    add_column :tool_proxies, :string, :string
```
Comment (abutterf) [inline db/migrate/20170424144902_add_submission_service_url_to_tool_proxy.rb]: Why are we calling this field string?

File: app/controllers/event_controller.rb (hunk 1 above)
Comment (westonkd) [inline app/controllers/event_controller.rb]: done

File: app/controllers/submissions_controller.rb (hunk 2 above)
Comment (westonkd) [inline app/controllers/submissions_controller.rb]: done

File: app/helpers/registration_helper.rb (hunk 3 above)
Comment (westonkd) [inline app/helpers/registration_helper.rb]: done

File: app/helpers/registration_helper.rb (hunk 4 above)
Comment (westonkd) [inline app/helpers/registration_helper.rb]: done

File: app/helpers/submissions_helper.rb (hunk 5 above)
Comment (westonkd) [inline app/helpers/submissions_helper.rb]: done

**Pull request merged**